vendor/paella_overrides/src/03_video_nodes.js

```diff
@@ -0,0 +1,1030 @@
+	}
+});
+
+Class ("paella.EmptyVideo", paella.VideoElementBase,{
+	initialize:function(id,stream,left,top,width,height) {
+		this.parent(id,stream,'div',left,top,width,height);
+	},
+
+	// Initialization functions
+	setPosterFrame:function(url) {},
+	setAutoplay:function(auto) {},
+	load:function() {return paella_DeferredRejected(new Error("no such compatible video player")); },
+	play:function() { return paella_DeferredRejected(new Error("no such compatible video player")); },
+	pause:function() { return paella_DeferredRejected(new Error("no such compatible video player")); },
+	isPaused:function() { return paella_DeferredRejected(new Error("no such compatible video player")); },
+	duration:function() { return paella_DeferredRejected(new Error("no such compatible video player")); },
+	setCurrentTime:function(time) { return paella_DeferredRejected(new Error("no such compatible video player")); },
+	currentTime:function() { return paella_DeferredRejected(new Error("no such compatible video player")); },
+	setVolume:function(volume) { return paella_DeferredRejected(new Error("no such compatible video player")); },
+	volume:function() { return paella_DeferredRejected(new Error("no such compatible video player")); },
+	setPlaybackRate:function(rate) { return paella_DeferredRejected(new Error("no such compatible video player")); },
+	playbackRate: function() { return paella_DeferredRejected(new Error("no such compatible video player")); },
+	unFreeze:function() { return paella_DeferredRejected(new Error("no such compatible video player")); },
+	freeze:function() { return paella_DeferredRejected(new Error("no such compatible video player")); },
+	unload:function() { return paella_DeferredRejected(new Error("no such compatible video player")); },
+	getDimensions:function() { return paella_DeferredRejected(new Error("no such compatible video player")); }
+});
+
+Class ("paella.videoFactories.EmptyVideoFactory", paella.VideoFactory, {
+	isStreamCompatible:function(streamData) {
+		return true;
+	},
+
+	getVideoObject:function(id, streamData, rect) {
+		return new paella.EmptyVideo(id, streamData, rect.x, rect.y, rect.w, rect.h);
+	}
+});
+
+Class ("paella.Html5Video", paella.VideoElementBase,{
+	_posterFrame:null,
+	_currentQuality:null,
+	_autoplay:false,
+	_streamName:null,
+
+	initialize:function(id,stream,left,top,width,height,streamName) {
+		this.parent(id,stream,'video',left,top,width,height);
+		this._streamName = streamName || 'mp4';
+		var This = this;
+
+		if (this._stream.sources[this._streamName]) {
+			this._stream.sources[this._streamName].sort(function (a, b) {
+				return a.res.h - b.res.h;
+			});
+		}
+
+		Object.defineProperty(this, 'video', {
+			get:function() { return This.domElement; }
+		});
+
+		this.video.preload = "auto";
+
+		function onProgress(event) {
+			// #DCE MATT-2046 Canvas IOS embedd waits at state 1
+			var readyStateLevel;
+			if (base.userAgent.system.iOS) {
```

Comment:
Are we only focusing on ios at the moment? Ignoring the preload attribute isn't unique to ios among mobile devices.